Wrappers/Python/test/test_Operator.py

```diff
@@ -372,11 +372,82 @@ def test_Norm(self):
         for n in [norm, norm2, norm3, norm4, norm5]:
             print ("norm {}", format(n))
 
-    
+    def test_ProjectionMap(self):
 
-        
+        # Check if direct is correct
+        ig1 = ImageGeometry(3,4)
+        ig2 = ImageGeometry(5,6)
+        ig3 = ImageGeometry(5,6,4)
+
+        # Create BlockGeometry
+        bg = BlockGeometry(ig1,ig2, ig3)
+        x = bg.allocate('random')
+
+        # Extract containers
+        x0, x1, x2 = x[0], x[1], x[2]
+
+        for i in range(3):
+
+            proj_map = ProjectionMap(bg, i)
 
+            # res1 is in ImageData from the X_{i} "ImageGeometry"
+            res1 = proj_map.direct(x)
 
+            # res2 is in ImageData from the X_{i} "ImageGeometry" using out
+            res2 = bg.geometries[i].allocate(0)
+            proj_map.direct(x, out=res2)
+
+            # Check with and without out
+            numpy.testing.assert_array_almost_equal(res1.as_array(), res2.as_array())
+
+            # Depending on which index is used, check if x0, x1, x2 are the same with res2
+            if i==0:            
+                numpy.testing.assert_array_almost_equal(x0.as_array(), res2.as_array())
+            elif i==1: 
+                numpy.testing.assert_array_almost_equal(x1.as_array(), res2.as_array())   
+            elif i==2:
+                numpy.testing.assert_array_almost_equal(x2.as_array(), res2.as_array())  
+            else:
+                pass      
+
+        # Check if adjoint is correct
+
+        bg = BlockGeometry(ig1, ig2, ig3, ig1, ig2, ig3)
+        x = ig1.allocate('random')
```

Comment:
same comment as above. I would not allocate randoms rather a constant array of not zeros.

Reply:
Fixed